packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -871,6 +874,12 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
       generator = this.resourceGeneratorMap.get(ITERATIVE_TABLE_GENERATOR);
     } else if (isSqlStrategy(strategy)) {
       generator = this.resourceGeneratorMap.get(SQL_LAMBDA_GENERATOR);
+    } else if (isImportedAmplifyDynamoDbModelDataSourceStrategy(strategy)) {
+      if (ctx.importedAmplifyDynamoDBTableMap && ctx.importedAmplifyDynamoDBTableMap[typeName]) {
```

Comment:
I'd prefer to handle the validations as invariants to the various checks we have to scatter through the code. E.g., when we introduced SQL support, we made strong assertions that every single model would have a DataSourceStrategy, and wrote our `is*` checks with that in mind.

What conditions would lead to this code being invoked with an invalid `importedAmplifyDynamoDBTableMap`? How can we catch those earlier so we can build in the strong assumptions through the rest of the code?

Perhaps a different internal structure work better: e.g., the imported strategy definition must be created with the mapped table name, so:

```typescript
export interface ImportedAmplifyDynamoDbModelDataSourceStrategy {
  readonly dbType: 'DYNAMODB';
  readonly provisionStrategy: 'IMPORTED_AMPLIFY_TABLE';
  readonly tableName: string;
}
```

Thus no import strategy could even be created without a valid mapping.

That might not be appropriate for the _public_ interface, if we want to allow a batch-oriented DX, but it would be completely appropriate for the _internal_ interface, where we want that strong consistency guarantee. But my initial thought is that even the public interface should be specified this way. That would be consistent with how we require SQL data sources to be specified--the strategy contains db connection information & db type, fully self-contained.